File: Year2code/StringPermutations/SPerms.java
import java.util.Scanner;
import java.util.Stack;
public class SPerms{
	public static void main(String[] args){
		Stack<String> s = new Stack<>();
		Stack<String> p = new Stack<>();
		Scanner sc = new Scanner(System.in);
		System.out.println("Input a String to have permutated.");
		String n = sc.nextLine();
		s.push(("+" + n));
		while(s.size() > 0){
			String t = s.pop();
			if(t.substring((t.length()-1),t.length()).equals("+")){
				p.push(t.substring(0,(t.length()-1)));
			}else{
				int i = 0;
				boolean g = true;
				while(g){
					if(t.substring(i,i+1).equals("+")){
						g = false;
					}else{
						i++;
					}
				}
				for(int j = i+1; j < t.length(); j++){
					s.push(t.substring(0,i)+t.substring(j,j+1)+t.substring(i,j)+t.substring(j+1,t.length()));
				}
			}
		}
		int c = 0;
		while(p.size() > 0){
			System.out.print(p.pop());
			c++;
			System.out.print(", ");

		}
		System.out.println(c);
	}
}

File: AdvancedPython/Sperms.py
n = input('Input a String to have permutated: ')
s= [('+'+n)]
p= []
while (len(s)>0):
	t = s.pop()
	if ( t[len(t)-1]=='+'):
		p.append(t[:len(t)-1])
	else:
		i=0
		g=True 
		while(g):
			if(t[i:i+1]=='+'):
				g=False
			else:
				i+=1
		for j in range(i+1,len(t)):
			s.append(t[0:i]+t[j:j+1]+t[i:j]+t[j+1:len(t)])
c=0
while(len(p)>0):
	print(p.pop()+',', end=' ')
	c+=1
print(c)
				